### PR TITLE
Refactor FlushToken to take unique_ptr as argument and remove const_cast

### PR DESCRIPTION
**Pull request by sduzh**
Summary: remove the `const_cast` and pass memory table by unique_ptr instead of shared_ptr

### CODE REVIEW

**Comment by sduzh** [comment]
@chaoyli @DeepThinker666 @wyb @imay PTAL

File: be/src/storage/memtable_flush_executor.cpp [hunk 1]
```diff
@@ -83,20 +94,18 @@ void FlushToken::_flush_memtable(std::shared_ptr<vectorized::MemTable> memtable)
...
             .build(&_flush_pool);
 }
 
-// NOTE: we use SERIAL mode here to ensure all mem-tables from one tablet are flushed in order.
-OLAPStatus MemTableFlushExecutor::create_flush_token(std::unique_ptr<FlushToken>* flush_token,
-                                                     ThreadPool::ExecutionMode execution_mode) {
-    *flush_token = std::make_unique<FlushToken>(_flush_pool->new_token(execution_mode));
-    return OLAP_SUCCESS;
+std::unique_ptr<FlushToken> MemTableFlushExecutor::create_flush_token(ThreadPool::ExecutionMode execution_mode) {
+    SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(nullptr);
```
Comment (imay) [inline be/src/storage/memtable_flush_executor.cpp:107]: Better to add some comment why change mem tracker in this function.

Comment (sduzh) [reply be/src/storage/memtable_flush_executor.cpp:106]: It's unused now, I removed it.

**Review by imay — DISMISSED** [review]
LGTM

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**